docker/scripts/build_and_clean.sh

```diff
@@ -3,11 +3,16 @@
 function build_and_clean() {
     local ccache_dir=$1
     local install_base=$2
+    local cuda_image=$3
+
+    local cmake_args=" -Wno-dev --no-warn-unused-cli"
+    if [ "$cuda_image" = true ]; then
+        # cspell: ignore DFORCE
+        cmake_args="$cmake_args -DFORCE_CUDA=1"
+    fi
 
     du -sh "$ccache_dir" && ccache -s &&
-        colcon build --cmake-args \
-            " -Wno-dev" \
-            " --no-warn-unused-cli" \
```

Comment:
For future reference:

# Explanation of Removed CMake Arguments

## 1. `-Wno-dev`
- **Purpose:** Suppresses "developer warnings" from CMake, which are aimed at improving `CMakeLists.txt` or related files.
- **Effect of Removal:** Developer warnings will now appear, possibly cluttering the output. These warnings may help identify issues in the build configuration.

## 2. `--no-warn-unused-cli`
- **Purpose:** Suppresses warnings about unused command-line arguments passed to CMake.
- **Effect of Removal:** Warnings will be displayed for unused arguments, helping catch typos or redundant inputs but potentially adding noise.

### Summary of Impact
- **Pros:** Can uncover latent issues in the build configuration.
- **Cons:** Increases verbosity, making it harder to spot critical errors.

Reply:
Oh these are not being removed but moved to the `build_and_clean.sh`.